src/HasMedia/ProcessMediaTrait.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Brackets\Media\HasMedia;
+
+use Brackets\Media\Exceptions\FileCannotBeAdded\FileIsTooBig;
+use Brackets\Media\Exceptions\FileCannotBeAdded\TooManyFiles;
+use Illuminate\Http\File;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Facades\Validator;
+use Spatie\MediaLibrary\Exceptions\FileCannotBeAdded;
+use Spatie\MediaLibrary\Models\Media;
+use Spatie\MediaLibrary\Models\Media as MediaModel;
+
+/**
+ * @property-read boolean $autoProcessMedia
+ */
+trait ProcessMediaTrait
+{
+    protected $autoProcessMedia = true;
```

Comment:
@dejwCake I was thinking about extracting one more trait - AutoProcessMediaTrait that would include only:
```
/**
     * Setup to auto process during saving
     */
    public static function bootHasMediaCollectionsTrait(): void
    {
        static::saving(static function ($model) {
            /** @var self $model */
            $model->processMedia(collect(request()->only($model->getMediaCollections()->map->getName()->toArray())));
        });
    }
```